wordpress-mobile/WordPress-iOS
# Tracks: Fix currently selected screen value when Login is set as root

Fixes n/a

The app fires a tracking event everytime it moves to the background. One of the parameters is the `lastVisibleScreen`, which is populated by values representing:
- The current/selected view controller displayed on the tab bar controller.
- A post editor screen that is currently presented.
- A login screen that is currently presented.

However, the value for this field will be "Blog List" (or the default screen displayed by the tab bar controller) when actually, the Login screen is shown as the root view controller. This also applies to the upcoming Jetpack migration flow screens. 

The PR fixes this by checking the type of the current `rootViewController`, and returns that instead.

#### Relation to the Content Migration Flow

While unrelated at a glance, the issue affects Content Migration Flow. More specifically, the state of the app post-migration. Here's why:

<details>
<summary>Click for more details</summary>

- The app collects tracking data when the app is backgrounded. This logic resides in `trackApplicationClosed` in `WPAnalytics`. The method invokes a stored callback, `lastVisibleScreenCallback`, which calls `currentlySelectedScreen`.
- This method calls the `WPTabBarController.sharedInstance`, which causes the tab bar to prematurely initialize (along with all the three main navigation controllers for My Site, Reader, and Notifications) while the Login screen is still displayed as root view controller.
  - `NSFetchRequestController`s then starts to fetch the objects, but since the user hasn't logged in yet, the API request fails and returns an empty result.
- After importing the data, the Notifications tab shows an empty result due to the previous fetch attempt (before logging in).

Note that this specific fetch issue above is avoidable when the `WPAccountDefaultWordPressComAccountChanged` notification is fired (in fact, we will do that), but we don't know what other problems would arise when the tab bar is instantiated prematurely. 

</details>

## To test

- Launch WP or JP app.
- On the login screen, minimize the app.
- Verify that `🔵 Tracked: application_closed <last_visible_screen: Login View, time_in_app: 6, time_in_main_reader: 0, time_in_reader_filtered_list: 0, time_in_reader_post: 0>` is called.

## Regression Notes
1. Potential unintended areas of impact
Should be none.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
